apps/web/pages/settings/my-account/out-of-office/index.tsx

```diff
@@ -39,18 +40,58 @@ export type BookingRedirectForm = {
   toTeamUserId: number | null;
   reasonId: number;
   notes?: string;
+  uuid?: string | null;
 };
 
-const CreateOutOfOfficeEntryModal = ({
+export type outOfOfficeEntryData = {
+  uuid: string;
+  selectedReason: number | null;
+  profileRedirect: boolean;
+  selectedMember: number | null;
+  notes: string | null;
+  start: Date;
+  end: Date;
+};
+
+const CreateOrEditOutOfOfficeEntryModal = ({
```

Comment:
I don't like this; there are too many things in a single file, making the code look very messy. For new contributors, it’s difficult to understand what is going on.

Suggestion:
Extract this dialog into a different component. It's not a good practice to put all components in one file, as it becomes very hard to make changes later.

I've mentioned this before, and I'll mention it again: handle things conditionally. For example, when creating an 'ooo' event, the temp property should be empty or null initially. However, when updating, temp should hold the value saved before. I don’t understand why there are so many setValue calls and so many things happening. There are repeated forms, properties, and handlers being used.

Tagging @hariombalhara and @Udit-takkar if they have any better ideas